src/Modules/Backend/Domain/Authentication/TwoFactorAuthenticationFactory.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace ForkCMS\Modules\Backend\Domain\Authentication;
+
+use ForkCMS\Modules\Extensions\Domain\Module\ModuleName;
+use ForkCMS\Modules\Extensions\Domain\Module\ModuleSettings;
+
+class TwoFactorAuthenticationFactory
+{
+    public function __construct(private readonly ModuleSettings $moduleSettings)
+    {
+    }
+
+    public function secret(): string
+    {
+        return $this->moduleSettings->get(
+            ModuleName::fromString('Backend'),
+            '2fa_secret',
+            ''
```

Comment:
Why not `null`?
Or maybe the default symfony secret?

Reply:
The default symfony secret isn't long enough, and it is required for 2fa. During install off the module a new key is generated for this purpose.